src/slam_solver.h

```diff
@@ -0,0 +1,22 @@
+/* Extends ScanSolver to allow different solvers to publish graph visualizations differently */
+#ifndef SLAM_SOLVER_H
+#define SLAM_SOLVER_H
+
+#include <open_karto/Mapper.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <string>
+
+namespace karto {
+
+class SLAMSolver : public karto::ScanSolver
+{
+  public:
+    virtual void publishGraphVisualization(visualization_msgs::MarkerArray &marray)=0;
```

Comment:
This doesn't seem to publish. `createVisualizationMarkers` would fit better. It can return the marker array instead of accepting marray as reference. In the current interface it is not clear if marray will be cleared before markers are added to it.